src/Resources/TSIGKey.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Exonet\Powerdns\Resources;
+
+class TSIGKey
+{
+    /**
+     * The name of the key.
+     *
+     * @var string
+     */
+    private $name;
+
+    /**
+     * The ID for this key, used in the TSIGkey URL endpoint.
+     *
+     * @var string
+     */
+    private $id;
+
+    /**
+     * The algorithm of the TSIG key.
+     *
+     * @var string
+     */
+    private $algorithm;
+
+    /**
+     * The Base64 encoded secret key, empty when listing keys. MAY be empty when POSTing to have the server generate the key material.
+     *
+     * @var string
+     */
+    private $key;
+
+    /**
+     * Set to "TSIGKey".
+     *
+     * @var string
+     */
+    private $type = 'TSIGKey';
+
+    /**
+     * Record constructor.
+     *
+     * @param string $content Optional content to set.
+     */
+    public function __construct(?array $content = null)
+    {
+        if ($content) {
+            $this->setName($content['name'] ?? '');
+            $this->setId($content['id'] ?? '');
+            $this->setAlgorithm($content['algorithm'] ?? '');
+            $this->setKey($content['key'] ?? '');
+            $this->setType('TSIGKey');
+        }
+    }
+
+    /**
+     * Get set to "TSIGKey".
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * Set set to "TSIGKey".
+     *
+     * @param string $type Set to "TSIGKey"
+     *
+     * @return self
+     */
+    public function setType(string $type)
+    {
+        $this->type = $type;
+
+        return $this;
+    }
```

Comment:
Should it be possible to change the type?

Reply:
the type is changeable, but seems pretty fixed in the documentation i'll remove the setter :)